tsl/src/chunk_copy.c

```diff
@@ -625,6 +811,197 @@ chunk_copy(Oid chunk_relid, const char *src_node, const char *dst_node, bool del
 	}
 	PG_END_TRY();
 
-	/* Cleanup and delete the catalog entry */
-	chunk_copy_cleanup(&cc);
+	/* Finish up and delete the catalog entry */
+	chunk_copy_finish(&cc);
+}
+
+static ScanTupleResult
+chunk_copy_activity_tuple_found(TupleInfo *ti, void *const data)
+{
+	ChunkCopy **cc = data;
+
+	*cc = STRUCT_FROM_SLOT(ti->slot, ti->mctx, ChunkCopy, FormData_chunk_copy_activity);
+	return SCAN_CONTINUE;
+}
+
+static ChunkCopy *
+chunk_copy_activity_get(int32 activity_id, const char *operation_id)
+{
+	ScanKeyData scankeys[1];
+	ChunkCopy *cc = NULL;
+	int indexid;
+	MemoryContext old, mcxt;
+
+	/* Objects need to be in long lived context */
+	mcxt =
+		AllocSetContextCreate(PortalContext, "chunk copy cleanup activity", ALLOCSET_DEFAULT_SIZES);
+	old = MemoryContextSwitchTo(mcxt);
+
+	/* search via any one of the provided inputs */
+	if (activity_id > 0)
+	{
+		ScanKeyInit(&scankeys[0],
+					Anum_chunk_copy_activity_idx_id,
+					BTEqualStrategyNumber,
+					F_INT4EQ,
+					Int32GetDatum(activity_id));
+		indexid = CHUNK_COPY_ACTIVITY_PKEY_IDX;
+	}
+	else if (operation_id != NULL)
+	{
+		ScanKeyInit(&scankeys[0],
+					Anum_chunk_copy_activity_idx_operation_id,
+					BTEqualStrategyNumber,
+					F_NAMEEQ,
+					DirectFunctionCall1(namein, CStringGetDatum(operation_id)));
+		indexid = CHUNK_COPY_ACTIVITY_UKEY_OPERATION_IDX;
+	}
+	else
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("invalid chunk copy activity identifier")));
+
+	ts_catalog_scan_one(CHUNK_COPY_ACTIVITY,
+						indexid,
+						scankeys,
+						1,
+						chunk_copy_activity_tuple_found,
+						AccessShareLock,
+						CHUNK_COPY_ACTIVITY_TABLE_NAME,
+						&cc);
+
+	/*
+	 * If a valid entry is returned then fill up the rest of the fields in the
+	 * ChunkCopy structure
+	 */
+	if (cc)
+	{
+		cc->mcxt = mcxt;
+		cc->chunk = ts_chunk_get_by_id(cc->fd.chunk_id, true);
+		cc->stage = NULL;
+
+		/* Was the chunk compressed in the meanwhile? */
+		if (ts_chunk_is_compressed(cc->chunk))
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("\"%s\" is a compressed remote chunk. Chunk copy/move cleanup not "
+							"supported"
+							" currently on compressed chunks",
+							get_rel_name(cc->chunk->table_id))));
```

Comment:
Should not have several sentences in main error message according to the error style guide. I suggest moving additional information to an `errdetail` or `errhint`. For example:

```suggestion
			ereport(ERROR,
					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
					 errmsg("copy chunk cleanup not supported on compressed chunks"),
					 errhint("Decompress chunk \"%s\" before cleaning up the copy chunk operation.",
							get_rel_name(cc->chunk->table_id))));
```

Reply:
Removed this error message now as per earlier discussions.